plugin/evm/vm_test.go

```diff
@@ -3509,16 +3518,18 @@ func TestFeeManagerRegressionMempoolMinFeeAfterRestart(t *testing.T) {
 	require.Equal(t, newHead.Head.Hash(), common.Hash(blk.ID()))
 }
 
-func restartVM(vm *VM, sharedDB database.Database, genesisBytes []byte, appSender commonEng.AppSender, finishBootstrapping bool) (*VM, error) {
-	vm.Shutdown(context.Background())
+func restartVM(tvm *testVM, tvmConfig testVMConfig) (*testVM, error) {
+	if err := tvm.vm.Shutdown(context.Background()); err != nil {
+		return nil, err
+	}
 	restartedVM := &VM{}
-	vm.ctx.Metrics = metrics.NewPrefixGatherer()
-	err := restartedVM.Initialize(context.Background(), vm.ctx, sharedDB, genesisBytes, nil, nil, []*commonEng.Fx{}, appSender)
+	tvm.vm.ctx.Metrics = metrics.NewPrefixGatherer()
+	err := restartedVM.Initialize(context.Background(), tvm.vm.ctx, tvm.db, []byte(tvmConfig.genesisJSON), []byte(tvmConfig.upgradeJSON), []byte(tvmConfig.configJSON), []*commonEng.Fx{}, tvm.appSender)
 	if err != nil {
 		return nil, err
 	}
```

Comment:
```suggestion
	if err != nil {
		if shutdownErr := restartedVM.Shutdown(context.Background()); shutdownErr != nil {
			log.Error("failed to shutdown VM after initialization failure", "err", shutdownErr)
		}
		return nil, err
	}
```

Reply:
It will take care of releasing resources with `defer`.